app/services/doc_auth/lexis_nexis/error_generator.rb

```diff
@@ -56,48 +59,89 @@ class ErrorGenerator
+        elsif user_error_count > 1
+          # Simplify multiple errors into a single error for the user
+          error_fields = errors.keys
+          if error_fields.length == 1
+            case error_fields.first
+            when ID
+              errors[ID] = Set[general_error(false)]
+            when FRONT
+              errors[FRONT] = Set[I18n.t('doc_auth.errors.lexis_nexis.multiple_front_id_failures')]
+            when BACK
+              errors[BACK] = Set[I18n.t('doc_auth.errors.lexis_nexis.multiple_back_id_failures')]
+            end
+          elsif error_fields.length > 1
+            return { GENERAL => [general_error(liveness_enabled)] } if error_fields.include?(SELFIE)
+
+            # If we don't have a selfie error don't give the message suggesting retaking selfie.
+            return { GENERAL => [general_error(false)] }
+          end
+        end
+
+        errors.transform_values(&:to_a)
+      end
+      # rubocop:enable Metrics/PerceivedComplexity
+
+      def self.get_error_messages(liveness_enabled, response_info)
         errors = Hash.new { |hash, key| hash[key] = Set.new }
 
         if response_info[:DocAuthResult] != 'Passed'
-          response_info[:Alerts]&.each do |alert|
+          response_info[:Alerts][:failed]&.each do |alert|
             alert_msg_hash = TRUE_ID_MESSAGES[alert[:name].to_sym]
 
             if alert_msg_hash.present?
-              # Don't show alert errors if the DocAuthResult has passed
-              # With liveness turned on DocAuthResult can pass but the liveness check fails
-              if alert[:result] != 'Passed'
-                # We should log the counts of failing alerts that are given to users for analytics
-                # AM: Log to Ahoy/Cloudwatch
-                errors[alert_msg_hash[:type]].add(I18n.t(alert_msg_hash[:msg_key]))
-              end
-              # We always want to make sure any unknown alerts that come through are noted
-              # AM: Log to Ahoy/Cloudwatch
+              errors[alert_msg_hash[:type]].add(I18n.t(alert_msg_hash[:msg_key]))
             end
           end
-        elsif liveness_checking_enabled && response_info[:PortraitMatchResults].present?
-          # Only bother to look for selfie_results if ID Auth was successful
-          if response_info[:PortraitMatchResults].dig(:FaceMatchResult) != 'Pass'
-            errors[:selfie].add(I18n.t('doc_auth.errors.lexis_nexis.selfie_failure'))
-            # Should probably log if selfie results isn't populated when we expect it to be?
-          end
         end
 
-        if errors.empty?
-          e = UnknownTrueIDError.new('TrueID failure escaped without useful errors')
-          # AM: Log to Ahoy/Cloudwatch
-          NewRelic::Agent.notice_error(e, { custom_params: { response_info: response_info } })
+        pm_results = response_info[:PortraitMatchResults]
+        if liveness_enabled && pm_results.present? && pm_results.dig(:FaceMatchResult) != 'Pass'
```

Comment:
the `.present?` check and the `.dig` are a lil redundant, I bet we could simplify:
```suggestion
        pm_results = response_info[:PortraitMatchResults] || {}
        if liveness_enabled && pm_results.dig[:FaceMatchResult] != 'Pass'
```

Reply:
I believe the issue I was running into there was that if it wasn't there the check would still pass because nil != 'Pass'.

I'll noodle on this a bit.

Reply:
Finally noodled a bit about this and you're right. I don't think the  present? is needed.